src/Equilibrium/types.jl

```diff
@@ -0,0 +1,140 @@
+- `InverseRunInput`:     Internal data structure for the inverse solver.
+- `PlasmaEquilibrium`:   The final, user-facing output object.
+"""
+module Types
+
+export EquilInput, DirectRunInput, InverseRunInput, PlasmaEquilibrium
+
+"""
+    EquilInput(...)
+
+A mutable struct holding the input parameters that control the equilibrium
+reconstruction process. These parameters are typically set by the user.
+
+## Arguments:
+- `eq_filename`: A string with the path to the equilibrium file (e.g., g-file).
+- `eq_type`: A string specifying the equilibrium file format (e.g., "efit").
+- `jac_type`: A string specifying the Jacobian type for coordinate transformation.
+- `psilow`: The lower bound of the normalized psi grid for output (typically 0.0).
+- `psihigh`: The upper bound of the normalized psi grid for output (typically 1.0).
+- `mpsi`: The number of grid points in the `psi` direction for the output.
+- `mtheta`: The number of grid points in the `theta` direction for the output.
+
+## Keyword Arguments:
+- `newq0`: Target q-value on axis. If non-zero, triggers q-profile revision (Default: 0.0).
+"""
+mutable struct EquilInput
+    eq_filename::String
+    eq_type::String
+    jac_type::String
+    power_bp::Int
+    power_r::Int
+    power_b::Int
+    grid_type::String
+    psilow::Float64
+    psihigh::Float64
+    mpsi::Int
+    mtheta::Int
+    newq0::Float64
+
+    function EquilInput(
+        eq_filename::String, eq_type::String, jac_type::String,
+        psilow::Float64, psihigh::Float64, mpsi::Int, mtheta::Int;
+        newq0::Float64=0.0
+    )
+        new(eq_filename, eq_type, jac_type, 0, 0, 0, "ldp", psilow, psihigh, mpsi, mtheta, newq0)
+    end
+end
+
+"""
+    DirectRunInput(...)
+
+A container struct that bundles all necessary inputs for the `direct_run` function.
+It is created by the `_read_efit` function after parsing the raw equilibrium file
+and preparing the initial splines.
+
+## Fields:
+- `equil_input`: The original `EquilInput` object.
+- `sq_in`
+        # x value: psin
+        # Quantity 1: F = R*Bt  [m T]
+        # Quantity 2: Pressure (non-negative) [nt^2 / m^2 * mu0 = T^2]
+        # Quantity 3: q-profile 
+        # Quantity 4: sqrt(psi_norm)
+- `psi_in`:
+        # x, y value: R, Z [m]
+        # z value : poloidal flux adjusted to be zero at the boundary [Weber/radian]
+            # 1. ψ(R,Z) = ψ_boundary - ψ(R,Z)
+            # 2. if ψ = ψ * sign(ψ(centerR,centerZ))
+- `rmin`: Minimum R-coordinate of the computational grid [m].
+- `rmax`: Maximum R-coordinate of the computational grid [m].
+- `zmin`: Minimum Z-coordinate of the computational grid [m].
+- `zmax`: Maximum Z-coordinate of the computational grid [m].
+- `psio`: The total flux difference `abs(ψ_axis - ψ_boundary)` [Weber / radian].
+"""
+mutable struct DirectRunInput
+    equil_input::EquilInput
+    sq_in::Any       # 1D profile spline (CubicSplineType)
+    psi_in::Any      # 2D flux spline (BicubicSplineType)
```

Comment:
Putting `Any` or any abstract type in a struct will lead to bad performance. Not critical to fix now, but if you know the concrete, use it. If you're not sure, you want to use [parametric types](https://docs.julialang.org/en/v1/manual/types/#Parametric-Types)